packages/devtools-map-bindings/src/utils.js

```diff
@@ -1,35 +1,31 @@
 // @flow
 
 import type { MappedScopeBindings } from "debugger-html";
+const { default: generate } = require("babel-generator");
+
+const babylon = require("babylon");
+const { default: traverse } = require("babel-traverse");
 
 function replaceOriginalVariableName(
   expression: string,
   generatedScopes: MappedScopeBindings[]
 ): string {
-  // JavaScript indetifier is a complex thing: it can contain '$', '\', ZWJ, any
-  // unicode letter or number. Simplifing that to latin character set for now.
-  // For simplicity, search only first characters of the expression to find an
-  // identifier/variable name.
-  const possibleVarNameRegex = /^([$\w]+)/;
-  const matchedVarName = possibleVarNameRegex.exec(expression);
-  if (!matchedVarName) {
-    return expression;
-  }
+  const ast = babylon.parse(expression, {
+    sourceType: "module",
+    plugins: ["jsx", "flow", "objectRestSpread"],
+  });
 
-  const originalName = matchedVarName[1];
-  // The generatedScopes sorted in inner-to-outer scope order, finding first.
-  const foundScope = generatedScopes.find(
-    ({ bindings }) => originalName in bindings
-  );
-  if (!foundScope) {
-    return expression;
-  }
+  traverse(ast, {
+    Identifier(path) {
+      const { node: { name } } = path;
+      const foundScope = generatedScopes.find(({ bindings }) => name in bindings);
+      if (foundScope) {
+        path.node.name = foundScope.bindings[name];
+      }
+    },
+  });
 
-  // Replace original name (which will be at the beginning of the expression)
-  // with found generated name.
-  const generatedName = foundScope.bindings[originalName];
-  const expressionWoOriginalName = expression.substring(originalName.length);
-  return `${generatedName}${expressionWoOriginalName}`;
+  return generate(ast, { concise: true, compact: true }).code.replace(";", "");
```

Comment:
what does this do: `.replace(";", "");`

Reply:
`babel-generator` appears append a `;` on the generated output i.e. `"first*2"` becomes `"n*2;"`. To match the test assertions you provided, I removed the trailing `;`; thinking about this some more, there's probably a more standardized approach :)

Reply:
there may be a case when user typed ';' as part of the expression, e.g. `str.replace(";", "")`. it's better to use regexp there: `.replace(/;$/, "")`.